project_template/inc/az/exampleshortname/operations.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/** @file operations.h */
+#pragma once
+#include <stdint.h>
+#include <az/exampleshortname/export.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+/** @brief add two numbers together
+ *  @param[in] a
+ *  @param[in] b
+ *  @return @p a + @p b
+ */
+AZ_EXAMPLESHORTNAME_EXPORT int32_t az_exampleshortname_add_two_numbers(int32_t a, int32_t b);

```

Comment:
>AZ_EXAMPLESHORTNAME_EXPORT [](start = 0, length = 26)

Question: what is this for? Is it for abstracting declspecs?
In the past in the IotHub SDK for C team when we had to have exports for dynamic libraries we did that exclusively through .def files in order to not introduce extra artifacts in the code.
And we did not need other declspecs TBH as we shied away from most platform specific things.

Reply:
this is very likely to change due to feedback from Robert.

On windows, this will add __declspec(dllimport) which makes calling the function a little faster.

Reply:
As additional information that may be relevant if we want to maintain .def files for windows, it looks like there is an available equivalent for other platforms:

```
gcc -Wl,--version-script=symbols.map
```

https://stackoverflow.com/questions/792195/gnu-linker-alternative-to-version-script-to-list-exported-symbols-at-the-comm

I would encourage this direction if strong ABI stability is a goal.